### PR TITLE
RB-EditPost#8

### PR DESCRIPTION
**Pull request by robert-brunner**
# Description
Trello Card-https://trello.com/c/31vi9q8L/8-8-edit-a-post
I have completed the Edit feature on the post menu in regards to ticket `EditPost#8`.  When the user clicks on the `Post` menu, they will see a list of possible posts to choose from.   There will be a **pencil** icon between the **eye** and **trash-can**.  
Edit Pencil
When the user clicks it they should be redirected to this **Edit Form**:
Edit Form
Upon clicking the Save button at the bottom of the **Edit Form**, the user will be redirected to the post in question.  The desired changes should be present on that post. 
Post With edit button
Near the middle of the current post page there will be another **pencil**.  If the user clicks that **pencil**, it will redirect them back to the **Edit Form**.  

# Testing Instructions

1)`git fetch -a`
2) `git checkout <RB-EditPost#8>`
3) If you don't see the most recent version of my files, `git pull origin <RB-EditPost#8>`
4) In Visual Studio (if you have not created the TabloidMVC database), 
- Open the SQL Server Object Explorer
- In the Server Object Explorer window, find and right-click the SQLEXPRESS tab.
- Click new query, in the window that pops up, copy the code from the following files and click play to run the queries.
- Fill the columns and rows: https://github.com/NewForce-Cohort-6/tabloidmvc-the-tootsie-rolls/blob/main/SQL/01_TabloidMVC_Create_DB.sql
- Seed data: https://github.com/NewForce-Cohort-6/tabloidmvc-the-tootsie-rolls/blob/main/SQL/02_TabloidMVC_Seed_Data.sql
5) Open Visual Studio, click "Open a project or solution", find and click "TabloidMVC.sln"
6) Run the program through the debugger
7) If needed, log in using "admin@example.com"
8) After logging in click the **Posts** option.
9) Scroll thru the potential posts and click the **pencil** on whichever post you wish to edit.  
10)The page will redirect you to an **Edit Form** where you can make changes.  Once you are satasfied with your changes press the **Save** button. 
11) The page will redirect you to the current post.  If you click the edit **pencil** near the top of your post, it will once again redirect you to the edit form where you can make your changes.  Repeat step "10"

# Checklist:

- [ x] My code follows the style guidelines of this project
- [ x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ x] My changes generate no new warnings or errors
- [ x] I have added test instructions that prove my fix is effective or that my feature works

### CODE REVIEW

File: TabloidMVC/Controllers/PostController.cs
Comment (CaraNetzer) [inline TabloidMVC/Controllers/PostController.cs]: post keeps turning up null here because GetPublishedPostById is returning null (see next comment)

File: TabloidMVC/Repositories/PostRepository.cs
Comment (CaraNetzer) [inline TabloidMVC/Repositories/PostRepository.cs]: When I add a post, IsApproved defaults to 1, so when it searches the postRepository in the Edit function (prev comment) this query doesn't return any posts

File: TabloidMVC/Repositories/PostRepository.cs
Comment (CaraNetzer) [inline TabloidMVC/Repositories/PostRepository.cs]: Same thing is happening here, when I add posts to test and then try to view them on the Post tab, they don't show up because this query is looking for posts that are unapproved. Later on it will be important to only show posts that are approved, but rn it seems like everything is automatically set to 1/approved

**Comment by sarahebrooks12** [comment]
Testing instructions look good. Please resolve issues and have two testing approvals before merging

**Review by jacobjjude — APPROVED** [review]
🌟 Tested using instructions. Works as intended. As far as I can tell, previous issues have been resolved.

**Review by Trey-Chambers — APPROVED** [review]
Tested and working

**Pull request merged**
